### PR TITLE
Track signup modal panel nav button clicks

### PR DESCRIPTION
**Pull request by phoenixeliot**
Tracks when the user clicks the "next" button, when that next button succeeds (implicitly tracking when it fails), and when they click the "back" button.

### CODE REVIEW

File: app/views/core/CreateAccountModal/BasicInfoView.coffee [hunk 1]
```diff
@@ -176,13 +176,18 @@ module.exports = class BasicInfoView extends CocoView
       when 'teacher' then ['name', 'password', 'email', 'firstName', 'lastName']
       else ['name', 'password', 'email']
   
-  onClickBackButton: -> @trigger 'nav-back'
+  onClickBackButton: ->
+    if @signupState.get('path') is 'teacher'
```
Comment (differentmatt) [inline app/views/core/CreateAccountModal/BasicInfoView.coffee]: Why are we only tracking the teacher path?

File: app/views/core/CreateAccountModal/teacher/SetupAccountPanel.coffee [hunk 2]
```diff
@@ -9,6 +9,7 @@ SetupAccountPanel = Vue.extend
   mounted: ->
     @$store.dispatch('modal/createAccount')
     .catch (e) =>
+      window.tracker?.trackEvent 'CreateAccountModal Teacher SetupAccountPanel Error', category: 'Teachers', ['Mixpanel']
```
Comment (differentmatt) [inline app/views/core/CreateAccountModal/teacher/SetupAccountPanel.coffee]: Not sure we should capture errors in our analytics like this.  I don't think we do this anywhere else, so maybe a discussion of best practices is in order.

**Comment by differentmatt** [comment]
We should exclude Mixpanel instead of opt-in, unless we're definitely going to use Mixpanel to analyze this data.

**Comment by nwinter** [comment]
We still want to completely remove Mixpanel instead of doing any more work around it: https://app.asana.com/0/search/305235332244550/271985943788025

**Pull request merged**